apps/frontend/src/schemas/TimesheetSchema.tsx

```diff
@@ -1,32 +1,22 @@
 import { z } from "zod";
-import { RowSchema, ScheduledRowSchema, CommentSchema } from './RowSchema';
-
-// The status is either undefined, for not being at that stage yet, or 
-// contains the date and author of approving this submission 
-export const StatusEntryType = z.union(
-  [z.object({
-    Date: z.number(),
-    AuthorID: z.string()
-  }),
-  z.undefined()]);
-
-// Status type contains the four stages of the pipeline we have defined
-export const StatusType = z.object({
-  HoursSubmitted: StatusEntryType,
-  HoursReviewed: StatusEntryType,
-  ScheduleSubmitted: StatusEntryType,
-  Finalized: StatusEntryType
-});
+import { RowSchema, ScheduledRowSchema, CommentSchema } from "./RowSchema";
+import { StatusType } from "./StatusSchema";
 
 export const TimeSheetSchema = z.object({
-  TimesheetID: z.number(),  
-  UserID: z.string(), 
+  TimesheetID: z.number(),
+  UserID: z.string(),
   StartDate: z.number(),
-  Status: StatusType, 
-  CompanyID: z.string(), 
-  TableData: z.array(RowSchema), 
+  Status: StatusType,
+  CompanyID: z.string(),
+  TableData: z.array(RowSchema),
   ScheduleTableData: z.union([z.undefined(), z.array(ScheduledRowSchema)]),
-  WeekNotes: z.union([z.undefined(), z.array(CommentSchema)]), 
-}); 
+  WeekNotes: z.union([z.undefined(), z.array(CommentSchema)]),
+});
 
 export type TimeSheetSchema = z.infer<typeof TimeSheetSchema>;
+
+export enum TimesheetStatus {
```

Comment:
Would modifying our status schema to match the CardState give us the ability to remove this?

Reply:
Ah good catch. I can use that to replace the `CardState`

Other than the enum, I think this is an outdated schema that looks like it's not getting used anywhere. I'll delete the rest of this. We no longer track any kind of `status` enum in our backend data, but it could totally be something that gets calculated and added into the frontend/shared schema, which would take the work of calculating the current status of the timesheet out of the `SubmitCard` file (where it definitely shouldn't live). Should probably be another ticket to do that, and I'd want to finish the work of shared schemas first.

Reply:
I think that the `TimesheetStatus` enum will need to stay here though - I think it'll be helpful to have a calculated field that stores the current status, since our `Status` object includes all `StatusEntries`.

Reply:
ok makes sense